components/rebrandly/rebrandly.app.mjs

```diff
@@ -1,11 +1,105 @@
+import { axios } from "@pipedream/platform";
+
 export default {
   type: "app",
   app: "rebrandly",
-  propDefinitions: {},
+  propDefinitions: {
+    domain: {
+      label: "Domain",
+      description: "The domain to be used",
+      type: "string",
+      async options({ prevContext }) {
+        const domains = await this.getDomains({
+          params: {
+            last: prevContext.lastId,
+            orderBy: "createdAt",
+            orderDir: "asc",
+          },
+        });
+
+        return {
+          options: domains.map((domain) => ({
+            label: domain.fullName,
+            value: domain.id,
+          })),
+          context: {
+            lastId: domains.length > 0
+              ? domains[domains.length - 1].id
+              : undefined,
+          },
+        };
+      },
+    },
+    linkId: {
+      label: "Link Id",
+      description: "The ID of the link",
+      type: "string",
+      async options({ prevContext }) {
+        const links = await this.getLinks({
+          params: {
+            last: prevContext.lastId,
+          },
+        });
+
+        return {
+          options: links.map((link) => ({
+            label: link.title ?? link.destination,
+            value: link.id,
+          })),
+          context: {
+            lastId: links.length > 0
+              ? links[links.length - 1].id
+              : undefined,
```

Comment:
If marking this as `undefined`, wouldn't it start loading again from the first page?